src/libsyntax/parse/attr.rs

```diff
@@ -18,6 +23,40 @@ const DEFAULT_UNEXPECTED_INNER_ATTR_ERR_MSG: &str = "an inner attribute is not \
                                                      permitted in this context";
 
 impl<'a> Parser<'a> {
+    crate fn parse_arg_attributes(&mut self) -> PResult<'a, Vec<ast::Attribute>> {
+        let mut attrs: Vec<ast::Attribute> = Vec::new();
+        loop {
+            debug!("parse_arg_attributes: self.token={:?}", self.token);
+            match self.token {
+                token::Pound => {
+                    let attr = self.parse_attribute(false)?;
+                    let has_allowed_built_in_attrs = || {
+                        if attr.check_name("allow") || attr.check_name("cfg")
+                            || attr.check_name("cfg_attr") || attr.check_name("deny")
+                            || attr.check_name("forbid") || attr.check_name("warn")
+                        {
+                            true
+                        }
+                        else {
+                            BUILTIN_ATTRIBUTES.iter().all(|x| {
+                                !attr.check_name(x.0.as_str().get())
+                            })
+                        }
+                    };
+                    if !has_allowed_built_in_attrs() {
+                        let mut err = self.span_fatal(attr.span, "forbidden built-in attribute");
```

Comment:
Does this have to use `_fatal` here? Avoiding to abort compilation would be the better UX if possible.

Reply:
For functions without `self`, all attributes are analyzed. The problem with `self` is [here](https://github.com/rust-lang/rust/blob/4443957f272e304e083a8d98583e608d65a712aa/src/libsyntax/parse/parser.rs#L6308) where the function immediately returns if `self` is invalid.
I will try to figure something to parse every argument